sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -348,4 +363,99 @@ class HiveDDLSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       }
     }
   }
+
+  test("create/drop database - location") {
+    val catalog = sqlContext.sessionState.catalog
+    withTempDir { tmpDir =>
+      val dbName = "db1"
+      val tabName = "tab1"
+      val path = catalog.createDatabasePath(dbName, Option(tmpDir.toString))
+      val fs = new Path(path).getFileSystem(hiveContext.sessionState.hiveconf)
+      withTable(tabName) {
+        assert(tmpDir.listFiles.isEmpty)
+        sql(s"CREATE DATABASE $dbName Location '$tmpDir'")
+        val db1 = catalog.getDatabaseMetadata(dbName)
+        assert(db1 == CatalogDatabase(
+          dbName,
+          "",
+          path,
+          Map.empty))
+        sql("USE db1")
```

Comment:
After each test, we reset everything, as shown in the line:
https://github.com/apache/spark/pull/12081/files#diff-b7094baa12601424a5d19cb930e3402fR36